File: CodeCraft_Studio_App/app/src/main/java/com/example/codecraftstudioapp/models/DetailPesanan.java
package com.example.codecraftstudioapp.models;

import com.google.gson.annotations.SerializedName;

public class DetailPesanan {
    @SerializedName("id_detail")
    private int idDetail;

    @SerializedName("id_menu")
    private int idMenu;

    @SerializedName("jumlah")
    private int jumlah;

    @SerializedName("harga_satuan")
    private double hargaSatuan;

    @SerializedName("kustomisasi")
    private String kustomisasi;

    @SerializedName("menu")
    private Menu menu;

    public int getIdDetail() { return idDetail; }
    public int getIdMenu() { return idMenu; }
    public int getJumlah() { return jumlah; }
    public double getHargaSatuan() { return hargaSatuan; }
    public String getKustomisasi() { return kustomisasi; }
    public Menu getMenu() { return menu; }
}

File: CodeCraft_Studio_Mobile/app/Models/DetailPesanan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class DetailPesanan extends Model
{
    use HasFactory;

    protected $table = 'detail_pesanan';
    protected $primaryKey = 'id_detail';
    public $timestamps = false;

    protected $fillable = [
        'id_pesanan',
        'id_menu',
        'jumlah',
        'harga_satuan',
        'kustomisasi'
    ];

    public function pesanan()
    {
        return $this->belongsTo(Pesanan::class, 'id_pesanan', 'id_pesanan');
    }

    public function menu()
    {
        return $this->belongsTo(Menu::class, 'id_menu', 'id_menu');
    }
}
